open_spiel/games/negotiation.cc

```diff
@@ -584,8 +584,12 @@ std::unique_ptr<State> NegotiationGame::DeserializeState(
     std::unique_ptr<State> state = NewInitialState();
     SPIEL_CHECK_EQ(lines.size(), 5);
     NegotiationState& nstate = static_cast<NegotiationState&>(*state);
-    // Take the chance action, but then reset the quantities.
+    // Take the chance action, but then reset the quantities. Make sure game's
+    // RNG state is not advanced during deserialization so copy it beforehand
+    // in order to be able to restore after the chance action.
+    std::mt19937* rng = new std::mt19937(*rng_);
```

Comment:
Can you use unique_ptr here? i.e. rng = absl::make_unique<std::mt19937>(*rng_);
rng_ = std::move(rng);